src/OmniSharp.Roslyn/OmniSharpWorkspace.cs

```diff
@@ -224,5 +271,11 @@ private void SaveDocumentText(DocumentId id, string fullPath, SourceText newText
                 _logger.LogError(e, $"Error saving document {fullPath}");
             }
         }
+
+        public bool IsCapableOfSemanticDiagnostics(Document document)
+        {
+            var documentId = GetDocumentId(document.FilePath);
+            return !_miscellanousFiles.Contains(documentId);
```

Comment:
Could we get rid of the _miscellaneousFiles hashset but checking to see if the projectId is the one we keep a reference to?